superset/views/sql_lab/views.py

```diff
@@ -161,6 +161,12 @@ def put(self, tab_state_id: int) -> FlaskResponse:
             return Response(status=403)
 
         fields = {k: json.loads(v) for k, v in request.form.to_dict().items()}
+        if client_id := fields.get("latest_query_id"):
+            query = db.session.query(Query).filter_by(client_id=client_id).one_or_none()
+            if not query:
+                return self.json_response(
+                    {"error": f"Unknown client_id: {client_id}"}, status=400
```

Comment:
This is a good improvement, thanks! I would recommend a slight change to the error message just to say that the update failed, rather than that the client_id is unknown in case anyone tries to use this endpoint to find a client_id that is valid.

Reply:
Oki, updated.